build/sync/plan.yml

```diff
@@ -0,0 +1,28 @@
+checks:
+  - type: repo_is_clean
+    params:
+      repo: template
+  - type: repo_is_clean
+    params:
+      repo: plugin
+actions:
+  - paths:
+      - build
+    actions:
+      - type: overwrite_directory
+        params:
+          create: true
+  - paths:
+      - Makefile
+      - go.mod
```

Comment:
Should this ready be here? What about repo specific dependencies?

Reply:
@hanzei I have `go.mod` here as it made including `mattermost-server`@commit easiest to do, rather than copy&pasting etc. I resolve any conflicts after the overwrite has been done. I can take it out if it doesn't make sense.

We should probably discuss how we want to use the tool in general, as we have been just using it however we see fit. Also, committing the `plan.yml` used is just a convention I thought would be useful, though idk how useful it really is.

Reply:
I quite like committing `plan.yml` as it makes it easier to sync the repo as someone else alread thought about the question what should be synced.

But I think `go` will do the "right thing" most of the time. See also https://github.com/mattermost/mattermost-plugin-starter-template/pull/86#discussion_r466993380